pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -193,7 +198,29 @@ func (exp *flowExporter) initFlowExporter(collectorAddr string, collectorProto s
 	// TODO: This code can be further simplified by changing the go-ipfix API to accept
 	// collectorAddr and collectorProto instead of net.Addr input.
 	var expInput exporter.ExporterInput
-	if collectorProto == "tcp" {
+	if exp.enableTLSToFlowAggregator {
+		// if CA certificate, client certificate and key do not exist during initialization,
+		// it will retry to obtain the credentials in next export cycle
+		caCert, err := getCACert()
+		if err != nil {
+			return err
+		}
+		clientCert, clientKey, err := getClientCertKey(exp.k8sClient)
+		if err != nil {
+			return err
+		}
```

Comment:
I think these errors should be wrapped to add context.

e.g. for the first one `return fmt.Errorf("cannot retrieve CA cert: %v", err)`